docs/user/ppl/cmd/parse.rst

```diff
@@ -0,0 +1,97 @@
+=============
+parse
+=============
+
+.. rubric:: Table of contents
+
+.. contents::
+   :local:
+   :depth: 2
+
+
+Description
+============
+| The ``parse`` command parses a text field using a regular expression and append the result to the search result.
+
+
+Syntax
+============
+parse <field> <regular-expression>
+
+* field: mandatory. The field must be a text field.
+* regular-expression: mandatory. The regular expression used to extract new fields from given text field. If a new field name already exists, it will replace the original field.
+
+Regular Expression
+==================
+
+The regular expression is used to match the whole text field of each document with Java regex engine. Each captured group in the expression will become a new field with type to be ``STRING`` by default. The field type can be casted using the following syntax for a captured group: ``(?<fieldName[fieldType]>regex)``. List of available field types:
```

Comment:
Could we add [cast](https://github.com/opensearch-project/sql/blob/main/docs/user/dql/functions.rst#cast) function in PPL? By default, parse extract field of String type.

Reply:
removed the casting logic in this PR, PPL supports cast now (#433)